### DISCUSSION TITLE
Does Hyperlink support `target=_new`?

### DISCUSSION THREAD

**Discussion by vbeffa** [emilk/egui#938] [Q&A]
I need to open links in a new tab/window.

**Comment by parasyte** [comment]
(FWIW, when clicking links in a desktop app, the page _always_ opens in a new tab.)

It isn't going to be possible inside a browser/WASM build unless the `webbrowser` crate is updated to pass `"_blank"` as the `windowName`/`target` to [`window.open()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/open):

https://github.com/amodm/webbrowser-rs/blob/1ce016d3eccdf974be89c415e7afc5b225c96fe2/src/lib.rs#L173-L186

There was an issue filed on their repo, but it was closed by the requester claiming that `window.open()` already does what they want. (I don't think it does.) https://github.com/amodm/webbrowser-rs/issues/31

**Reply by vbeffa** [reply]
It looks like

https://github.com/amodm/webbrowser-rs/blob/1ce016d3eccdf974be89c415e7afc5b225c96fe2/src/lib.rs#L178

would need to be changed to use

https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Window.html#method.open_with_url_and_target

instead of

https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Window.html#method.open_with_url

and allow a target to be passed into `webbrowser::open`.

**Reply by vbeffa** [reply]
I opened an issue: https://github.com/amodm/webbrowser-rs/issues/39.

When I have a moment I will submit a PR to address this.

**Comment by emilk** [comment]
@vbeffa Click the `Hyperlink` with the middle mouse button, or while holding down any modifier key (e.g. shift+click the link). It will open in a new tab (though maybe your browser will ask for permission first).

**Reply by vbeffa** [reply]
@emilk thank you. I'm aware of the mouse functionality, but I was looking for something to not require the users to have to use a different mouse button or hold a modifier key down. They will almost certainly forget to do this, and since the app we are developing requires a login, if the link opens in the same tab/window when you navigate back to the app it will require logging in again, which is less than ideal.

**Reply by emilk** [reply]
Sounds like an easy thing to add! https://github.com/emilk/egui/blob/master/egui/src/widgets/hyperlink.rs

**Comment by ctaggart** [comment]
This was fixed in August 2023 in https://github.com/emilk/egui/pull/3242.
